### PR TITLE
Button deprecation

### PR DESCRIPTION
**Pull request by pksjce**
Describe your changes here.

Move the old button to deprecated folder
Move the new button to the main folder
Add migration folder with appropriate docs

Closes # https://github.com/github/primer/issues/687

### Merge checklist

- [-] Added/updated tests
- [x] Added/updated documentation
- [x] Tested in Chrome
- [x] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: docs/content/migration-guide/Button.mdx [hunk 1]
```diff
@@ -0,0 +1,99 @@
+---
+title: Button migration guide
+---
```
Comment (siddharthkp) [inline docs/content/migration-guide/Button.mdx:3]: @rezrah Should this be in the changelog itself?

File: docs/src/@primer/gatsby-theme-doctocat/live-code-scope.js [hunk 2]
```diff
@@ -51,6 +52,7 @@ export default function resolveScope(metastring) {
     ...doctocatComponents,
     ...primerComponents,
     ...(metastring.includes('drafts') ? drafts : {}),
+    ...deprecated,
```
Comment (siddharthkp) [inline docs/src/@primer/gatsby-theme-doctocat/live-code-scope.js:55]: This unintentionally reverses the change done in https://github.com/primer/react/blob/next/v35.0.0/docs/src/@primer/gatsby-theme-doctocat/live-code-scope.js#L55

**Review by siddharthkp — CHANGES_REQUESTED** [review]
Using this checklist for review

### Checks
- [x] Docs working (Button counter example is broken)
- [ ] Tests passing (CI hasn't run yet)
- [x] Storybook working


### User facing (should prioritise for release)

- [x] Add changelog

API:
  - [x] deprecation: Move from src/index to src/deprecated
  - [x] deprecation: Add `@deprecated` JSDoc annotation
  - [x] promotion: Move from src/drafts to src/index

Docs:
  - [x] deprecation: Update status to deprecated in docs
  - [ ] deprecation: Add 'deprecated' to code examples (for ChoiceInputField)
  - [x] deprecation: Add deprecation notice in docs
  - [x] deprecation: Move to deprecated section in docs/nav.yml
  - [x] deprecation: Move to docs/deprecated/
  - [x] promotion: Move from docs/drafts/ to docs/
  - [x] promotion: Remove `drafts` from code examples
  - [x] promotion: Move out of drafts in docs/nav.yml
  - [x] promotion: Change name from Component2 to Component
  

### Internal maintenance (not blocking, can tackle in a new PR)

- [x] deprecation: Move deprecated component to src/deprecated/
- [x] promotion: Rename component from Component2 to Component
- [x] Update import path in related components

Tests
- [x] Move deprecated component to tests/deprecated/
- [x] Update import path in tests
- [x] Update test snapshots
- [x] promotion: Rename component from Label2 to Label

Storybook:
- [ ] Update title in story meta "Deprecated components"
- [ ] Move to stories/deprecated/
- [x] Update import path in stories of related components (can use deprecated too)

File: docs/content/migration-guide/Button.mdx (hunk 1 above)
Comment (rezrah) [reply docs/content/migration-guide/Button.mdx:3]: Yep, we agreed to merge the guides into release notes via changesets

File: docs/src/@primer/gatsby-theme-doctocat/live-code-scope.js (hunk 2 above)
Comment (pksjce) [reply docs/src/@primer/gatsby-theme-doctocat/live-code-scope.js]: Fixed this.

**Comment by pksjce** [comment]
Questions

What does the following mean?
1.  deprecation: Add 'deprecated' to code examples
2. Update title in story meta "Deprecated components"

**Comment by rezrah** [comment]
> Questions
> 
> What does the following mean?
> 
> 1. deprecation: Add 'deprecated' to code examples
> 2. Update title in story meta "Deprecated components"

I suspect it's..

1) in the mdx snippets you need to add the deprecated scope. E.g. "jsx live **deprecated**"
2) [Example here ](https://github.com/primer/react/blob/a8a3de110afc7edaf088f0773e76ce3edcf5c65d/src/stories/deprecated/ActionList.stories.tsx#L28)

@siddharthkp, feel free to add / correct if needed

File: docs/content/migration-guide/Button.mdx (hunk 1 above)
Comment (rezrah) [reply docs/content/migration-guide/Button.mdx]: @pksjce I can't see a changeset with the release notes at the moment? It's also why CI is failing atm.

**Comment by siddharthkp** [comment]
@rezrah's right on both of these

> in the mdx snippets you need to add the deprecated scope. E.g. "jsx live deprecated"

Without adding `deprecated` meta, the docs don't work because they can't find the component: https://primer-react-h3rprwugm-primer.vercel.app/react/deprecated/Buttons#kitchen-sink

**Review by siddharthkp — APPROVED** [review]
Preemptive approve because the checklist is almost done.

File: docs/content/migration-guide/Button.mdx (hunk 1 above)
Comment (pksjce) [reply docs/content/migration-guide/Button.mdx]: I added one and then with another commit it totally vanished! I'll add it back now.

**Comment by pksjce** [comment]
I think I added all thats needed. I will have a followup PR with a separate story and doc for `ButtonGroup` asap.

File: .changeset/many-roses-hammer.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
+---
+"@primer/react": patch
```
Comment (rezrah) [inline .changeset/many-roses-hammer.md:2]:
```suggestion
"@primer/react": major
```

File: .changeset/many-roses-hammer.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+## Change list
+
+### Button variants
+
+We now support a variant property which takes values `primary`, `invisible`, `outline` and `danger`
+
+#### Before
+
+```jsx
```
Comment (rezrah) [inline .changeset/many-roses-hammer.md:19]: Todo for later.. we'd want to make these tables, but we can do this in a follow up PR

**Review by rezrah — APPROVED** [review]

**Pull request merged**